python/bandpassUtils.py

```diff
@@ -0,0 +1,251 @@
+        defaultDirs[lens] = os.path.join(rootDir, 'components/camera', lens)
+    defaultDirs['filters'] = os.path.join(rootDir, 'components/camera/filters')
+    for mirror in ('mirror1', 'mirror2', 'mirror3'):
+        defaultDirs[mirror] = os.path.join(rootDir, 'components/telescope', mirror)
+    defaultDirs['atmosphere'] = os.path.join(rootDir, 'siteProperties')
+    return defaultDirs
+
+
+def buildVendorDetector(vendorDir, addLosses=True):
+    """
+    Assumes there is a file *QE.dat and *Losses.dat to represent the QE response and the Losses.
+    Returns a Bandpass object.
+    """
+    # Read the QE file.
+    qefile = glob(os.path.join(vendorDir, '*QE.dat'))
+    if len(qefile) != 1:
+        raise ValueError('Expected a single QE file in this directory, found: ', qefile)
+    qefile = qefile[0]
+    qe = Bandpass()
+    qe.readThroughput(qefile)
+    if addLosses:
+        # Read the Losses file(s). If more than one, multiply.
+        lossfiles = glob(os.path.join(vendorDir, '*Losses.dat'))
+        loss = Bandpass()
+        loss.readThroughputList(lossfiles)
+        wavelength, sb = qe.multiplyThroughputs(loss.wavelen, loss.sb)
+        qe.setBandpass(wavelength, sb)
+    return qe
+
+def buildGenericDetector(detectorDir, addLosses=True):
+    """
+    Combine detector throughputs from multiple vendors, using the minimum value of QE*losses at each wavelength.
+    Returns a bandpass object.
+    """
+    tmp = os.listdir(detectorDir)
+    vendorDirs = []
+    for t in tmp:
+        if os.path.isdir(os.path.join(detectorDir, t)):
+            vendorDirs.append(os.path.join(detectorDir, t))
+    #print "Using data from detector vendors: ", vendorDirs
+    sbAll = []
+    for vendorDir in vendorDirs:
+        qe = buildVendorDetector(vendorDir, addLosses=addLosses)
+        sbAll.append(qe.sb)
+    wavelen = qe.wavelen
+    sbMin = (np.array(sbAll)).min(axis=0)
+    qe.setBandpass(wavelen, sbMin)
+    return qe
+
+def buildFilters(filterDir, addLosses=True):
+    """
+    Assumes there are files [filtername]-bandResponse.dat, together with a filterLosses subdirectory containing loss files.
+    Returns a dictionary (keyed by filter name) of the bandpasses for each filter.
+    """
+    # Read the filter files.
+    filterfiles = glob(os.path.join(filterDir, '*-bandResponse.dat'))
+    filters = {}
+    for f in filterfiles:
+        fname = os.path.split(f)[1].split('-')[0]
+        filters[fname] = Bandpass()
+        filters[fname].readThroughput(f)
+    if addLosses:
+        # Read and multiply the losses.
+        lossfiles = glob(os.path.join(filterDir, 'filterLosses', '*.dat'))
+        loss = Bandpass()
+        loss.readThroughputList(lossfiles)
+        for f in filters:
+            wavelen, sb = filters[f].multiplyThroughputs(loss.wavelen, loss.sb)
+            filters[f].setBandpass(wavelen, sb)
+    return filters
+
+def savitzky_golay(y, window_size=31, order=3, deriv=0, rate=1):
```

Comment:
This whole function was lifted from Chuck's ipython notebook. I actually don't know what it's supposed to do, other than perhaps smooth the wavelength response of the borosilicate glass throughput curve (and I don't know why you have to do that). I have a request to Chuck to clarify and to explain, but otherwise this whole function is just here because it looked like I needed it (see makeLens*.ipynb -- no comments on this method there either). Perhaps @cclaver can fill in some more information on this PR?